src/Microsoft.Health.Fhir.SqlServer/Features/Search/SqlServerSearchService.cs

```diff
@@ -232,6 +230,34 @@ private async Task<SearchResult> SearchImpl(SearchOptions searchOptions, bool hi
                             rawResource = await streamReader.ReadToEndAsync();
                         }
 
+                        // if we have more columns, it means sort expressions were added.
+                        if (reader.FieldCount > 9)
+                        {
+                            try
+                            {
+                                // if sort is requested asc keep the first result for continuation token
+                                if (matchCount == 0 && searchOptions.Sort?[0].sortOrder == SortOrder.Ascending)
+                                {
+                                    sortExpr = reader.GetValue("SortExpr") as DateTime?;
```

Comment:
the `if` and `else if` blocks look identical...

Reply:
They only look identical, but we are interested mainly in the `SortOrder.Ascending` and `SortOrder.Descending` which differentiates the two blocks.

Reply:
Not the conditions, but the statement blocks are identical (the "then" parts). Use an `||` in the condition instead of duplicating

Reply:
This section was removed based on our discussion